modules/core/src/lib/utils.js

```diff
@@ -87,3 +87,35 @@ export function generatePointsParallelToLinePoints(
 
   return [p3.geometry.coordinates, p4.geometry.coordinates];
 }
+
+export function convertFeatureListToFeatureCollection(featuresList: Array<any>): FeatureCollection {
+  const features = featuresList.map(feature => {
+    // Turf functions do not preserve properties nested inside of a FeatureCollection feature's
+    // geometry. This logic moves the nested geometry properties up to the parent feature.
+    const featureObject: any = {
+      type: 'Feature',
+      geometry: feature,
+      properties: feature.properties || {}
+    };
+    return featureObject;
+  });
+  return {
+    type: 'FeatureCollection',
+    features
+  };
+}
+
+export function convertFeatureCollectionToFeatureList({ features }: FeatureCollection): Array<any> {
+  return features.map(({ geometry, properties }) => {
+    // Turf functions do not preserve a properties value of _internalIndex: 0
+    // It is important for the properties._internalIndex value to be present to prevent
+    // a strange mass duplication bug.
+    if (properties) {
+      geometry.properties = properties;
+      if (!properties._internalIndex) {
```

Comment:
I think we can get rid of the need for `internalIndex`. And the reason is because the ordering of indexes in `selectedFeatureIndexes` is the same as the ordering of features in the `FeatureCollection` we hand to turf. And all these turf operations (rotate, scale, translate) should not be altering the ordering of indexes.

I simplified the implementation for `TranslateHandler` to show what I'm talking about. See my branch [here](https://github.com/austintang/nebula.gl/compare/master...uber:clay/review-160?expand=1).

Reply:
This is much cleaner! However is it safe to rely on the order of `selectedIndexes` and `movedFeatures` or is order already accounted for?
```
    const movedFeatures = turfTransformTranslate(
      this._geometryBeforeTranslate,
      distanceMoved,
      direction
    );

    const selectedIndexes = this.getSelectedFeatureIndexes();
    for (let i = 0; i < selectedIndexes.length; i++) {
      const selectedIndex = selectedIndexes[i];
      const movedFeature = movedFeatures.features[i];
      updatedData = updatedData.replaceGeometry(selectedIndex, movedFeature.geometry);
    }
```

Reply:
@supersonicclay I used your simplified implementation in my latest diff. Let me know if you see any other issues.